src/HDF5.jl

```diff
@@ -1144,19 +1150,29 @@ function names(x::HDF5Attributes)
 end
 
 # iteration by objects
-# "next" opens new objects, "done" closes the old one. This prevents resource leaks.
-start(parent::Union{HDF5File, HDF5Group}) = Any[1, nothing]
-function done(parent::Union{HDF5File, HDF5Group}, iter::Array{Any})
-    obj = iter[2]
-    if !(obj === nothing)
-        close(obj)
+@static if VERSION < v"0.7.0-DEV.5126"
+    # "next" opens new objects, "done" closes the old one. This prevents resource leaks.
+    start(parent::Union{HDF5File, HDF5Group}) = Any[1, nothing]
+    function done(parent::Union{HDF5File, HDF5Group}, iter::Array{Any})
+        obj = iter[2]
+        if !(obj === nothing)
+            close(obj)
+        end
+        iter[1] > length(parent)
+    end
+    function next(parent::Union{HDF5File, HDF5Group}, iter)
+        iter[2] = h5object(h5o_open_by_idx(checkvalid(parent).id, ".", H5_INDEX_NAME, H5_ITER_INC, iter[1]-1, H5P_DEFAULT), parent)
+        iter[1] += 1
+        (iter[2], iter)
+    end
+else
+    function iterate(parent::Union{HDF5File, HDF5Group}, iter = (1,nothing))
+        n, prev_obj = iter
+        prev_obj ≢ nothing && close(prev_obj)
```

Comment:
this symbol is equivalent to `!(obj === nothing)`?

Reply:
yes (or, equivalently, `!==`)